src/components/CustomCollection/CustomBookmark/CustomBookmarkForm.tsx

```diff
@@ -0,0 +1,162 @@
+import React, { useEffect } from 'react'
+import { useFormik, FormikState, FormikValues } from 'formik'
+import * as yup from 'yup'
+import {
+  Form,
+  ButtonGroup,
+  Button,
+  Label,
+  TextInput,
+  Alert,
+  ErrorMessage,
+} from '@trussworks/react-uswds'
+
+type CustomBookmarkFormProps = {
+  onSave: (url: string, label: string) => void
+  onCancel: () => void
+  onDelete?: () => void
+  label?: string
+  url?: string
+  showAddWarning?: boolean
+  nameInputRef: React.RefObject<HTMLInputElement>
+  urlInputRef: React.RefObject<HTMLInputElement>
+}
+export const CustomBookmarkForm = ({
+  onSave,
+  onCancel,
+  onDelete,
+  label,
+  url,
+  showAddWarning,
+  nameInputRef,
+  urlInputRef,
+}: CustomBookmarkFormProps) => {
+  // #TODO: Integrate Formik into our forms following the
+  // wrapper component pattern used in other Truss projects
+
+  const formik = useFormik({
+    initialValues: {
+      // We have to provide label and url as possible initial values
+      // in case a user is editing
+      // #TODO this might break tests i already 'fixed'
+      bookmarkLabel: label || '',
+      bookmarkUrl: url || '',
+    },
+    validateOnChange: false, // #TODO check with design
+    validateOnBlur: true, // #TODO check with design
```

Comment:
@MKDale @malworks Do you all have guidance on handling in-line validation? We can do any combination of the following:

* Validate/show errors on submit
* Validate/show errors on blur of a field
* Validate/show errors when a field changes

Happy to do a quick demo too so y'all can see what it looks like